zipkin/propagation.go

```diff
@@ -28,11 +28,12 @@ func (p Propagator) Inject(
 		return opentracing.ErrInvalidCarrier
 	}
 
-	textMapWriter.Set("x-b3-traceid", strconv.FormatUint(sc.TraceID(), 16))
+	// TODO this needs to change to support 128bit IDs
```

Comment:
@szdavid92 I think there may be a bug here. According to the B3 spec, the length of the encoded ID should be 16 or 32 hex characters, i.e. it should be 0-padded, which FormatUint doesn't do.

In contrast, Jaeger's encoding prefers unpadded hex strings.